lasagne/layers/conv.py

```diff
@@ -794,8 +793,9 @@ def convolve(self, input, **kwargs):
             subsample=self.stride, border_mode=border_mode,
             filter_flip=not self.flip_filters)
         output_size = self.output_shape[2:]
-        if any(s is None for s in output_size):
-            output_size = self.get_output_shape_for(input.shape, True)[2:]
+        if (not isinstance(self.output_size, T.TensorVariable) and
+                any(s is None for s in output_size)):
+            output_size = self.get_output_shape_for(input.shape)[2:]
```

Comment:
That's also wrong. When `self.output_size` is `None`, the code should do the same as before (i.e., before your PR). I think my suggestion is fully correct. Of course, you need to keep the existing `output_size = self.output_shape[2:]` before the part that's modified here.